sdk/storage/azure-storage-blob/azure/storage/blob/_blob_service_client.py

```diff
@@ -771,19 +760,20 @@ def get_blob_client(
                 "Please use 'BlobProperties.name' or any other str input type instead.",
                 DeprecationWarning
             )
-        try:
-            container_name = container.name
-        except AttributeError:
-            container_name = container
-        try:
-            blob_name = blob.name
-        except AttributeError:
+            if blob.name is not None:
```

Comment:
Also, we shouldn't need this? Isn't `BlobProperties.name` already not optional?

Reply:
In my infinite wisdom I undid this work when glazing through the whole `_models` file.... My initial change when I did it for the GitHub issue is the most accurate because that was the time that I was cross-checking which values usually came populated.

I will revert this change in BSC, I will also revert the change in models, and I see that for basically the whole models file I need to do the same process that I did previously for just BlobProperties 😞